### PR TITLE
fix(vscode): Make codeful setting WORKFLOW_CODEFUL_ENABLED source of truth, remove redundant telemetry/context

### PR DESCRIPTION
**Pull request by andrew-eldridge**
## Commit Type

- [ ] feature - New functionality
- [x] fix - Bug fix
- [ ] refactor - Code restructuring without behavior change
- [ ] perf - Performance improvement
- [ ] docs - Documentation update
- [ ] test - Test-related changes
- [ ] chore - Maintenance/tooling

## Risk Level

- [x] Low - Minor changes, limited scope
- [ ] Medium - Moderate changes, some user impact
- [ ] High - Major changes, significant user/system impact

## What & Why
Updates 'isCodeful' checks to use the WORKFLOW_CODEFUL_ENABLED app setting as source of truth rather than csproj content, which is a less reliable/standard indicator.

## Impact of Change

- **Users**: May fix codeful project detection in some edge case scenarios
- **Developers**: Removes code for redundant telemetry and extension context
- **System**: N/A

## Test Plan

- [x] Unit tests added/updated
- [ ] E2E tests added/updated
- [x] Manual testing completed
- [ ] Tested in: 

## Contributors
@andrew-eldridge

### CODE REVIEW

File: apps/vs-code-designer/src/app/utils/codeless/startDesignTimeApi.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ import { Uri, window, workspace, type MessageItem } from 'vscode';
 import { findChildProcess } from '../../commands/pickFuncProcess';
 import find_process from 'find-process';
 import { getChildProcessesWithScript } from '../findChildProcess/findChildProcess';
-import { isCodefulProject } from '../codeful';
+import { hasCodefulSdkReference } from '../codeful';
 import {
```
[the remark this answers is not in the document]
Comment (andrew-eldridge) [reply apps/vs-code-designer/src/app/utils/codeless/startDesignTimeApi.ts:57]: need to check csproj when local.settings.json is not available, which is the case here

File: apps/vs-code-designer/src/constants.ts [hunk 2]
```diff
@@ -98,6 +98,7 @@ export const customConnectorResourceGroupNameKey = 'CUSTOM_CONNECTOR_RESOURCE_GR
 export const workflowSubscriptionIdKey = 'WORKFLOWS_SUBSCRIPTION_ID';
 export const workflowTenantIdKey = 'WORKFLOWS_TENANT_ID';
 export const workflowManagementBaseURIKey = 'WORKFLOWS_MANAGEMENT_BASE_URI';
+export const workflowCodefulEnabledKey = 'WORKFLOW_CODEFUL_ENABLED';
 export const workflowAppApiVersion = '2018-11-01';
```
[the remark this answers is not in the document]
Comment (andrew-eldridge) [reply apps/vs-code-designer/src/constants.ts:102]: updated to use const

**Review by lambrianmsft — APPROVED** [review]
